### PR TITLE
Add ability to promote a specific label

### PR DESCRIPTION
**Pull request by max-mironov**
See https://github.com/Microsoft/code-push/issues/267 for more details

### CODE REVIEW

File: cli/script/command-parser.ts [hunk 1]
```diff
@@ -356,6 +356,7 @@ var argv = yargs.usage(USAGE_PREFIX + " <command>")
             .example("promote MyApp Staging Production", "Promotes the latest release within the \"Staging\" deployment of \"MyApp\" to \"Production\"")
             .example("promote MyApp Staging Production --des \"Production rollout\" -r 25", "Promotes the latest release within the \"Staging\" deployment of \"MyApp\" to \"Production\", with an updated description, and targeting only 25% of the users")
             .option("description", { alias: "des", default: null, demand: false, description: "Description of the changes made to the app with this release. If omitted, the description from the release being promoted will be used.", type: "string" })
+            .option("label", { alias: "l", default: null, demand: false, description: "Label of the source release that will be taken. If omitted, the latest release being promoted will be used.", type: "string" })
```
Comment (silhouettes) [inline cli/script/command-parser.ts]: Could you update the `README.md` in the CLI directory also?

File: cli/script/command-executor.ts [hunk 2]
```diff
@@ -1071,6 +1071,7 @@ function promote(command: cli.IPromoteCommand): Promise<void> {
     var packageInfo: PackageInfo = {
         appVersion: command.appStoreVersion,
         description: command.description,
+        label: command.label,
```
Comment (silhouettes) [inline cli/script/command-executor.ts]: Can you update the `log("Successfully ...` line below to mention the label that was promoted (only if it was specified by the user)?

**Comment by max-mironov** [comment]
@silhouettes, I've updated Documentation and log message as requested above.

File: cli/README.md [hunk 3]
```diff
@@ -751,6 +752,10 @@ We recommend that all users take advantage of the automatically created `Staging
 
 This is the same parameter as the one described in the [above section](#description-parameter), and simply allows you to override the description that will be used for the promoted release. If unspecified, the new release will inherit the description from the release being promoted.
 
+### Label parameter
+
+This optional parameter allows you to pick up the corresponding label from source deployment and promote to destination deployment. If unspecified, the latest label will be used from the release being promoted..
```
Comment (silhouettes) [inline cli/README.md]: Slight tweak to expression:

> This optional parameter allows you to pick the specified label from the source deployment and promote it to the destination deployment. If unspecified, the latest release on the source deployment will be promoted.

**Comment by max-mironov** [comment]
@silhouettes, ok, got it, also I've updated README text.

**Comment by keertimaan** [comment]
Thank you for quickly making the change and enabling the feature. 👍

**Comment by runticle** [comment]
Is this doable using the SDK too? Can't find any info about it. Super useful feature for us 👍

**Pull request merged**